static/src/javascripts/projects/common/modules/preferences/main.js

```diff
@@ -73,8 +75,122 @@ define([
+                            });
+                    });
+                };
+
+                var unsubscribe = function () {
+                    // Disable the button so it can't be changed while
+                    // we process the permission request
+                    pushButton.disabled = true;
+
+                    return navigator.serviceWorker.ready.then(function (serviceWorkerRegistration) {
+                        return serviceWorkerRegistration.pushManager.getSubscription()
+                            .then(function (subscription) {
+                                if (!subscription) {
+                                    // No subscription? Reset UI
+                                    pushButton.disabled = false;
+                                    updateState({ pushEnabled: false });
+                                } else {
+                                    return subscription.unsubscribe()
+                                        .then(function () {
+                                            pushButton.disabled = false;
+                                            updateState({ pushEnabled: false });
+
+                                            return sendSubscription(subscription, { delete: true });
+                                        });
+                                }
+                            });
+                    });
+                };
+
+                var sendSubscription = function (subscription, options) {
+                    options = options || {};
+                    // TODO: var url
+
+                    var mobileNotificationsWebHost = config.page.pushNotificationsHost;
+
+                    if (!mobileNotificationsWebHost) {
+                        throw new Error('No notifications host found');
+                    }
+
+                    /*global fetch, Headers*/
+                    return fetch(mobileNotificationsWebHost + '/web/subscription', {
+                        method: options.delete ? 'DELETE' : 'POST',
+                        headers: new Headers({ 'Content-Type': 'application/json' }),
+                        // TODO: Support deprecated subscriptionId (now part of endpoint)
+                        body: JSON.stringify({
+                            subId: subscription.subscriptionId,
+                            endpoint: subscription.endpoint.replace('/' + subscription.subscriptionId, '')
+                            // TODO: edition based notifications
+                            //edition: config.page.edition
+                        })
+                    });
+                };
+
+                pushButton.addEventListener('click', function () {
+                    if (isPushEnabled) {
+                        unsubscribe();
+                    } else {
+                        subscribe();
+                    }
+                });
+
+                navigator.serviceWorker.register('/2015-05-19-service-worker.js')
```

Comment:
We had planned to move this to a separate page
(`/preferences/notifications`), so I've just done that. We simply won't
promote it until we are ready.

On Tue, 19 May 2015 at 14:58 Fabio Crisci <notifications@github.com> wrote:

> In static/src/javascripts/projects/common/modules/preferences/main.js
> <https://github.com/guardian/frontend/pull/9214#discussion_r30601055>:
>
> > +                            endpoint: subscription.endpoint.replace('/' + subscription.subscriptionId, '')
> > +                            // TODO: edition based notifications
> > +                            //edition: config.page.edition
> > +                        })
> > +                    });
> > +                };
> > +
> > +                pushButton.addEventListener('click', function () {
> > +                    if (isPushEnabled) {
> > +                        unsubscribe();
> > +                    } else {
> > +                        subscribe();
> > +                    }
> > +                });
> > +
> > +                navigator.serviceWorker.register('/2015-05-19-service-worker.js')
>
> So the plan is to switch it on on CODE? It'll still break that environment
> for all other people using it. Isn't it better to put it behind a 0% A/B
> test?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/9214/files#r30601055>.
>

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396